tests/pytests/test_debug_commands.py

```diff
@@ -362,29 +366,285 @@ def testSpecIndexesInfo(env: Env):
+            if remaining <= timeout_res_count * env.shardsCount:
+                # we don't know how many results are left in each shard, check only timeout
+                if env.isCluster():
+                    check_res_count = False
+                else:
+                    # in a single shard the next read will return EOF
+                    expected_results_per_iter = remaining
+            res, cursor = env.cmd('FT.CURSOR', 'READ', 'idx', cursor)
+            total_returned += len(res['results'])
+            if cursor == 0:
+                should_timeout = False
+
+            if check_res_count == False:
+                self.verifyWarning(res, f"AggregateDebug with cursor: iter: {iter}, total_returned: {total_returned}", should_timeout=should_timeout)
+            else:
+                self.verifyResultsResp3(res, expected_results_per_iter, f"AggregateDebug with cursor: iter: {iter}, total_returned: {total_returned}", should_timeout=should_timeout)
+            iter += 1
+        env.assertEqual(total_returned, self.num_docs, message=f"AggregateDebug with cursor: depletion took {iter} iterations")
+
+        # cursor count smaller than timeout count, expect no timeout
+        cursor_count = timeout_res_count // 2
+        cursor_query = [*basic_debug_query, 'WITHCURSOR', 'COUNT', cursor_count]
+        res, cursor = env.cmd(*cursor_query, 'LIMIT', 0, limit, *debug_params)
+        should_timeout = False
+        self.verifyResultsResp3(res, cursor_count, should_timeout=should_timeout, message="AggregateDebug with cursor count lower than timeout_res_count:")
+
+        self.StrictPolicy()
+
+    def testAggregateDebug(self):
+        self.AggregateDebug()
+
+    def testAggregateDebug_MT(self):
+        self.env.expect(config_cmd(), 'SET', 'WORKERS', 4).ok()
+        self.AggregateDebug()
+        self.env.expect(config_cmd(), 'SET', 'WORKERS', 0).ok()
+
+    # compare results of regular query and debug query
+    def Sanity(self, cmd, query_params):
+        # avoid running this test in cluster mode, as it relies on the order of the results
+        skipTest(cluster=True)
+        env = self.env
+        results_count = 200
+        timeout_res_count = results_count - 1
+        query = ['FT.' + cmd, 'idx', '*', *query_params, 'LIMIT', 0, results_count]
+        debug_params = ["TIMEOUT_AFTER_N", timeout_res_count, "DEBUG_PARAMS_COUNT", 2]
+
+        # expect that the first timeout_res_count of the regular query will be the same as the debug query
+        regular_res = env.cmd(*query)
+        debug_res = env.cmd(debug_cmd(), *query, *debug_params)
+        self.verifyResultsResp3(debug_res, timeout_res_count, f"{cmd} Sanity: compare regular and debug results", should_timeout=True)
+
+        for i in range(timeout_res_count):
+            env.assertEqual(regular_res["results"][i], debug_res["results"][i], message=f"Sanity: compare regular and debug results at index {i}")
+
+    def testSearchSanity(self):
```

Comment:
Do you think that sanity tests with ft.profile are also needed? Also, do you feel that the test are also covering the internal cursors mechanism?

Reply:
this mechanism doesn't support FT.PROFILE
Didn't understand the second question